plugin.video.rtsplaytv/resources/lib/rtsplaytv.py

```diff
@@ -0,0 +1,1377 @@
+                break
+        if not json_chapter:
+            log('build_episode_menu: No chapter ID found \
+                for video_id %s' % video_id)
+            return
+
+        json_segment_list = json_chapter.get('segmentList', [])
+        if video_id == chapter_id:
+            if include_segments:
+                self.build_entry(json_chapter, banner)
+                for segment in json_segment_list:
+                    self.build_entry(segment, banner)
+            else:
+                if segment_option and json_segment_list:
+                    self.build_entry(json_chapter, banner, is_folder=True)
+                else:
+                    self.build_entry(json_chapter, banner)
+        else:
+            json_segment = None
+            for segment in json_segment_list:
+                if segment.get('id') == segment_id:
+                    json_segment = segment
+                    break
+            if not json_segment:
+                log('build_episode_menu: No segment ID found \
+                    for video_id %s' % video_id)
+                return
+            self.build_entry(json_segment, banner)
+
+    def build_entry(self, json_entry, banner=None, is_folder=False):
+        """
+        Builds an list item for a video or folder by giving the json part,
+        describing this video.
+
+        Keyword arguments:
+        json_entry -- the part of the json describing the video
+        banner     -- URL of the show's banner (default: None)
+        is_folder  -- indicates if the item is a folder (default: False)
+        """
+        log('build_entry')
+        title = json_entry.get('title')
+        vid = json_entry.get('id')
+        description = json_entry.get('description')
+
+        image = json_entry.get('imageUrl', '')
+        # RTS image links have a strange appendix '/16x9'.
+        # This needs to be removed from the URL:
+        image = image.strip('/16x9')
+
+        duration = int_or_none(json_entry.get('duration'), scale=1000)
+        if not duration:
+            duration = get_duration(json_entry.get('duration'))
+        date_string = str_or_none(json_entry.get('date'), default='')
+        dto = parse_datetime(date_string)
+        kodi_date_string = dto.strftime('%Y-%m-%d') if dto else None
+
+        list_item = xbmcgui.ListItem(label=title)
+        list_item.setInfo(
+            'video',
+            {
+                'title': title,
+                'plot': description,
+                'duration': duration,
+                'aired': kodi_date_string,
+            }
+        )
+        list_item.setArt({
+            'thumb': image,
+            'poster': image,
+            'banner': banner,
+        })
+        subs = json_entry.get('subtitleList', [])
+        if subs and SUBTITLES:
+            subtitle_list = [x.get('url') for x in subs if
+                             x.get('format') == 'VTT']
+            if not subtitle_list:
```

Comment:
you are passing an empty list to setSubtitles if no subtitles are found. Also note vtt is not exactly equivalent to srt, the timestamps may be different than the ones the kodi subrip parser supports.